x/pairing/keeper/delegator_rewards_test.go

```diff
@@ -248,12 +261,32 @@ func TestProviderRewardWithCommission(t *testing.T) {
 	require.Equal(t, balance, newBalance)
 
 	// the delegator should get the total rewards
-	resRewards, err = ts.QueryDualstakingDelegatorRewards(delegator1, provider, ts.spec.Index)
+	resRewards, err = ts.QueryDualstakingDelegatorRewards(delegator1, "", "")
 	require.Nil(t, err)
 	require.Equal(t, 1, len(resRewards.Rewards))
 	dReward = resRewards.Rewards[0]
 	expectedDRewardForRelay := mint.MulInt64(totalReward.Int64())
 	require.Equal(t, expectedDRewardForRelay.TruncateInt64(), dReward.Amount.Amount.Int64())
+
+	// claim delegator rewards and verify balance
+	claimRewardsAndVerifyBalance(ts, delegator1Acc.Addr, provider, ts.spec.Index)
+}
+
+func claimRewardsAndVerifyBalance(ts *tester, delegator sdk.AccAddress, provider string, chainID string) {
+	balance := ts.GetBalance(delegator)
+
+	ind := dualstakingtypes.DelegationKey(provider, delegator.String(), chainID)
+	reward, found := ts.Keepers.Dualstaking.GetDelegatorReward(ts.Ctx, ind)
+	require.True(ts.T, found)
+
+	_, err := ts.Servers.DualstakingServer.ClaimRewards(ts.Ctx, dualstakingtypes.NewMsgClaimRewards(delegator.String(), provider))
+	require.Nil(ts.T, err)
+
+	newBalance := ts.GetBalance(delegator)
+	require.Equal(ts.T, balance+reward.Amount.Amount.Int64(), newBalance)
+
+	_, found = ts.Keepers.Dualstaking.GetDelegatorReward(ts.Ctx, ind)
```

Comment:
maybe also (or instead) use the query again to verify that nothing is left?

Reply:
done. see https://github.com/lavanet/lava/commit/302b2bec531bd9308d142d746c327509bde9b547